flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/runtime/stream/jsonplan/JoinJsonPlanITCase.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.runtime.stream.jsonplan;
+
+import org.apache.flink.table.planner.factories.TestValuesTableFactory;
+import org.apache.flink.table.planner.utils.JsonPlanTestBase;
+import org.apache.flink.types.Row;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Test for join json plan. */
+public class JoinJsonPlanITCase extends JsonPlanTestBase {
+    private static final List<Row> small5FieldsRowData =
+            Arrays.asList(
+                    Row.of(1, 1L, 0, "Hallo", 1L),
+                    Row.of(2, 2L, 1, "Hallo Welt", 2L),
+                    Row.of(2, 3L, 2, "Hallo Welt wie", 1L),
+                    Row.of(3, 4L, 3, "Hallo Welt wie gehts?", 2L),
+                    Row.of(3, 5L, 4, "ABC", 2L),
+                    Row.of(3, 6L, 5, "BCD", 3L));
+    private static final List<Row> small3FieldsRowData =
+            Arrays.asList(
+                    Row.of(1, 1L, "Hi"), Row.of(2, 2L, "Hello"), Row.of(3, 2L, "Hello world"));
+
+    @Override
+    @Before
+    public void setup() throws Exception {
+        super.setup();
+        createTestValuesSourceTable("A", small3FieldsRowData, "a1 int", "a2 bigint", "a3 varchar");
```

Comment:
use `JavaScalaConversionUtil.toJava(TestData.smallData3())` to replace `small3FieldsRowData`